### PR TITLE
Enable Echelle Extraction to run through even when a bad order occurs

### PR DESCRIPTION
**Pull request by profxj**
As titled.

No new docs or tests, but I have been running on a Magellan/FIRE
dataset where skysub fails (due to an over-filled slit).

### CODE REVIEW

**Comment by profxj** [comment]
DevSuite tests have been launched in the cloud 
using https://github.com/pypeit/PypeIt-development-suite/pull/234

File: pypeit/core/skysub.py
Comment (kbwestfall) [inline pypeit/core/skysub.py:1221]: I think we should be using a mask here, not setting things to NaN and then checking for it.

File: pypeit/extraction.py
Comment (kbwestfall) [inline pypeit/extraction.py]: remove?

File: pypeit/extraction.py
Comment (kbwestfall) [inline pypeit/extraction.py]: remove?

File: pypeit/extraction.py
Comment (kbwestfall) [inline pypeit/extraction.py]: remove?

File: pypeit/pypeit.py
Comment (kbwestfall) [inline pypeit/pypeit.py:893]: Should we instead create a new mask set that is specific to object finding and/or extraction, instead of having to alter the SlitTraceSet for each spec2d file?

File: pypeit/slittrace.py
Comment (kbwestfall) [inline pypeit/slittrace.py]: I'm fine with this, but we might want to somehow warn users of this.  We might also want to set a horizon on how long we keep it.  Something like "To be deprecated in version 1.13".

File: pypeit/slittrace.py
Comment (kbwestfall) [inline pypeit/slittrace.py:271]: This should get updated once we propagate BitMaskArray usage.  These are DataContainer objects that do version checking.  We could update it to this kind if detailed bit checking, too.

**Review by kbwestfall — APPROVED** [review]
I think we should be using masks instead of nan checking, but everything else looks good.  Approving now, anyway.

File: pypeit/core/skysub.py
Comment (jhennawi) [inline pypeit/core/skysub.py:1229]: I agree with @kbwestfall here. Let's use a boolean rather than nan and nanmean

**Review by jhennawi — CHANGES_REQUESTED** [review]
Minor changes requested, i.e. associated with nan and nanmean.

File: pypeit/core/skysub.py
Comment (profxj) [reply pypeit/core/skysub.py:1221]: done

File: pypeit/core/skysub.py
Comment (profxj) [reply pypeit/core/skysub.py:1229]: done

File: pypeit/extraction.py
Comment (profxj) [reply pypeit/extraction.py]: removed

File: pypeit/extraction.py
Comment (profxj) [reply pypeit/extraction.py]: gone

File: pypeit/extraction.py
Comment (profxj) [reply pypeit/extraction.py]: done

File: pypeit/slittrace.py
Comment (profxj) [reply pypeit/slittrace.py:271]: sounds good

File: pypeit/pypeit.py
Comment (profxj) [reply pypeit/pypeit.py:893]: yes, that would be a better solution.

I think we should take a fresh look at masking for both the `spec2d` and `spec1d` outputs.
But not in this PR. ;)

File: pypeit/slittrace.py
Comment (profxj) [reply pypeit/slittrace.py]: added text saying we may remove in 1.13

**Review by jhennawi — APPROVED** [review]
Approved, but please pull in develop and then re-run all tests

**Comment by profxj** [comment]
DevSuite tests started

**Comment by profxj** [comment]
One test failed, but I have just fixed it and tested on my laptop:

Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  205 passed, 137 warnings in 208.59s (0:03:28) ---
--- PYTEST UNIT TESTS PASSED  118 passed, 242 warnings in 915.17s (0:15:15) ---
--- PYTEST VET TESTS FAILED  1 failed, 28 passed, 11 warnings in 786.33s (0:13:06) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 166/166 TESTS  ---
Testing Started at 2023-02-14T02:33:19.224262
Testing Completed at 2023-02-14T11:28:14.445975
Total Time: 8:54:55.221713

**Comment by profxj** [comment]
@badpandabear -- I made a minor change to the `collate1d` script 
  (and a test in the DevSuite) to handle multiple `slit_trace_bm`

If you could take a quick look before I merge, that would be great.  Thx

**Review by badpandabear — APPROVED** [review]
Looks good.

**Pull request merged**
